compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -3975,15 +3967,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         && field.name == sym::integer(index)
                     {
                         if let Some(&field_ty) = tys.get(index) {
-                            if self.tcx.features().offset_of_slice() {
-                                self.require_type_has_static_alignment(field_ty, expr.span);
-                            } else {
-                                self.require_type_is_sized(
-                                    field_ty,
-                                    expr.span,
-                                    ObligationCauseCode::Misc,
-                                );
-                            }
+                            self.require_type_has_static_alignment(field_ty, expr.span);
```

Comment:
Could you please add a comment onto `require_type_has_static_alignment` that notes that changing the behavior will extend the functionality of `offset_of` and is insta-stable?